rust/src/quic/logger.rs

```diff
@@ -30,14 +31,40 @@ fn log_template(tx: &QuicTransaction, js: &mut JsonBuilder) -> Result<(), JsonEr
             js.set_string("ua", &String::from_utf8_lossy(&ua))?;
         }
     }
-    js.open_array("cyu")?;
-    for cyu in &tx.cyu {
-        js.start_object()?;
-        js.set_string("hash", &cyu.hash)?;
-        js.set_string("string", &cyu.string)?;
+    if tx.cyu.len() > 0 {
+        js.open_array("cyu")?;
+        for cyu in &tx.cyu {
+            js.start_object()?;
+            js.set_string("hash", &cyu.hash)?;
+            js.set_string("string", &cyu.string)?;
+            js.close()?;
+        }
+        js.close()?;
+    }
+
+    if tx.extv.len() > 0 {
+        js.open_array("extensions")?;
+        for e in &tx.extv {
+            let mut etypes = String::with_capacity(32);
+            match write!(&mut etypes, "{}", e.etype) {
```

Comment:
Do you want detection also ?..

Reply:
in general yes. I guess we'd want parity with existing tls/tcp as a first step

Reply:
Ok... it is going to be a long ride then